kubernetes/template/DeploymentTemplates/script.sh

```diff
@@ -1,263 +1,386 @@
+	jq --arg authenticationMethod "client_certificate" '.properties.customCloudProfile.authenticationMethod=$authenticationMethod' | \
+	jq --arg SPN_CLIENT_ID $SPN_CLIENT_ID '.properties.servicePrincipalProfile.clientId = $SPN_CLIENT_ID' | \
+	jq --arg SPN_CLIENT_SECRET_KEYVAULT_ID $SPN_CLIENT_SECRET_KEYVAULT_ID '.properties.servicePrincipalProfile.keyvaultSecretRef.vaultID = $SPN_CLIENT_SECRET_KEYVAULT_ID' | \
+	jq --arg SPN_CLIENT_SECRET_KEYVAULT_SECRET_NAME $SPN_CLIENT_SECRET_KEYVAULT_SECRET_NAME '.properties.servicePrincipalProfile.keyvaultSecretRef.secretName = $SPN_CLIENT_SECRET_KEYVAULT_SECRET_NAME' >  $AZURESTACK_CONFIGURATION_TEMP
 
 else
-echo "Using AAD"
-
-sudo cat $AZURESTACK_CONFIGURATION | jq --arg SPN_CLIENT_ID $SPN_CLIENT_ID '.properties.servicePrincipalProfile.clientId = $SPN_CLIENT_ID' | \
-jq --arg SPN_CLIENT_SECRET $SPN_CLIENT_SECRET '.properties.servicePrincipalProfile.secret = $SPN_CLIENT_SECRET' >  $AZURESTACK_CONFIGURATION_TEMP
+	log_output -i "In AAD section to update (servicePrincipalProfile ) configurations."
+	sudo cat $AZURESTACK_CONFIGURATION | jq --arg SPN_CLIENT_ID $SPN_CLIENT_ID '.properties.servicePrincipalProfile.clientId = $SPN_CLIENT_ID' | \
+	jq --arg SPN_CLIENT_SECRET $SPN_CLIENT_SECRET '.properties.servicePrincipalProfile.secret = $SPN_CLIENT_SECRET' >  $AZURESTACK_CONFIGURATION_TEMP
+fi
 
-echo "Checking and moving "
+log_output -i "Checking and moving temp file data(servicePrincipalProfile) to main api model file."
 check_and_move_azurestack_configuration $AZURESTACK_CONFIGURATION_TEMP $AZURESTACK_CONFIGURATION
 
+log_output -i "Completed building API model file based on passed stamp information and other parameters."
+
+#####################################################################################
+# Section to login using Azure CLI.
+# https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack
+HYBRID_PROFILE=2018-03-01-hybrid
+log_output -i "Register to AzureStack cloud."
+retrycmd_if_failure 5 10 az cloud register -n $ENVIRONMENT_NAME --endpoint-resource-manager $TENANT_ENDPOINT --suffix-storage-endpoint $SUFFIXES_STORAGE_ENDPOINT --suffix-keyvault-dns $SUFFIXES_KEYVAULT_DNS --endpoint-active-directory-resource-id $ENDPOINT_ACTIVE_DIRECTORY_RESOURCEID --endpoint-active-directory $ENDPOINT_ACTIVE_DIRECTORY_ENDPOINT --endpoint-active-directory-graph-resource-id $ENDPOINT_GRAPH_ENDPOINT
+log_output -i "Set Azure stack environment."
+retrycmd_if_failure 5 10 az cloud set -n $ENVIRONMENT_NAME
+log_output -i "Update cloud profile."
+retrycmd_if_failure 5 10 az cloud update --profile $HYBRID_PROFILE
+
+if [ $IDENTITY_SYSTEM == "ADFS" ] ; then
+	log_output -i "Login to ADFS environment using Azure CLI."	
+	retrycmd_if_failure 5 10 az login --service-principal  -u $SPN_CLIENT_ID  -p $CERTIFICATE_LOCATION --tenant $TENANT_ID --debug 
```

Comment:
Is the `--debug` flag required?

Reply:
This is the recommend way at https://docs.microsoft.com/en-us/azure/azure-stack/user/azure-stack-version-profiles-azurecli2#connect-to-azure-stack